sdk/eventhub/azure-eventhub/tests/livetest/asynctests/test_send_async.py

```diff
@@ -155,15 +156,38 @@ async def test_send_with_partition_key_async(connstr_receivers):
         await client.send_batch(await client.create_batch())
 
     found_partition_keys = {}
+    reconnect_receivers = []
     for index, partition in enumerate(receivers):
-        received = partition.receive_message_batch(timeout=5000)
-        for message in received:
+        retry_total = 0
+        while retry_total < 3:
+            timeout = 5000 + retry_total * 1000
             try:
-                event_data = EventData._from_message(message)
-                existing = found_partition_keys[event_data.partition_key]
-                assert existing == index
-            except KeyError:
-                found_partition_keys[event_data.partition_key] = index
+                received = partition.receive_message_batch(timeout=timeout)
+                for message in received:
+                    try:
+                        event_data = EventData._from_message(message)
+                        existing = found_partition_keys[event_data.partition_key]
+                        assert existing == index
+                    except KeyError:
+                        found_partition_keys[event_data.partition_key] = index
+                if received:
+                    break
+                retry_total += 1
+            except uamqp.errors.ConnectionClose:
+                uri = "sb://{}/{}".format(live_eventhub['hostname'], live_eventhub['event_hub'])
+                sas_auth = uamqp.authentication.SASTokenAuth.from_shared_access_key(
+                    uri, live_eventhub['key_name'], live_eventhub['access_key'])
+
+                source = "amqps://{}/{}/ConsumerGroups/{}/Partitions/{}".format(
+                    live_eventhub['hostname'],
+                    live_eventhub['event_hub'],
+                    live_eventhub['consumer_group'],
+                    index)
+                partition = uamqp.ReceiveClient(source, auth=sas_auth, debug=True, timeout=0, prefetch=500)
+                reconnect_receivers.append(partition)
```

Comment:
since the ReceiveClients created in the fixture will also be closed in the fixture, I will just close the ReceiveClients that were created in this test here. Will make sure no more than one ReceiveClient is created and open at a time.

Reply:
sounds good to me